crates/ruff_linter/src/rules/perflint/rules/manual_list_comprehension.rs

```diff
@@ -405,7 +405,11 @@ fn convert_to_list_extend(
         "for"
     };
     let target_str = locator.slice(for_stmt.target.range());
-    let elt_str = locator.slice(to_append);
+    let elt_str = if to_append.is_generator_expr() {
+        format!("({})", locator.slice(to_append))
+    } else {
+        locator.slice(to_append).to_string()
+    };
     let generator_str = format!("{elt_str} {for_type} {target_str} in {for_iter_str}{if_str}");
```

Comment:
nit: this is arguably a bit less readable since it duplicates the long `format!` string, but it avoids allocating a separate `String` in the `format!`/`to_string` calls for `elt_str`, which I think makes it worth it.

```suggestion
    let elt_str = locator.slice(to_append);
    let generator_str = if to_append.is_generator_expr() {
        format!("({elt_str}) {for_type} {target_str} in {for_iter_str}{if_str}")
    } else {
        format!("{elt_str} {for_type} {target_str} in {for_iter_str}{if_str}")
    };
```